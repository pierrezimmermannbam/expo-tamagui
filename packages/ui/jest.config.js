// eslint-disable-next-line @typescript-eslint/no-var-requires
const baseNativeConfig = require('../../jest.config.native.js');

/** @type {import('@jest/types').Config.InitialOptions} */
module.exports = {
  ...baseNativeConfig,
  testMatch: ['<rootDir>/src/**/*.test.[jt]s?(x)'],
};
