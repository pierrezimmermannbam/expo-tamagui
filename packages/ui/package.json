{
  "name": "@my/ui",
  "version": "0.0.1",
  "sideEffects": [
    "*.css"
  ],
  "private": true,
  "types": "./types/index",
  "main": "src/index.tsx",
  "module:jsx": "src",
  "files": [
    "types",
    "dist"
  ],
  "scripts": {
    "build": "tamagui-build",
    "watch": "tamagui-build --watch"
  },
  "dependencies": {
    "@tamagui/animations-react-native": "latest",
    "@tamagui/font-inter": "latest",
    "@tamagui/react-native-media-driver": "latest",
    "@tamagui/shorthands": "latest",
    "@tamagui/themes": "latest",
    "@tamagui/toast": "latest",
    "tamagui": "latest"
  },
  "devDependencies": {
    "@tamagui/build": "latest",
    "@testing-library/react-native": "^12.4.1",
    "@types/jest": "^29.5.11",
    "jest": "^29.7.0"
  }
}
