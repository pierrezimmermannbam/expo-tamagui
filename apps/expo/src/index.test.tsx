import { Button } from '@my/ui';
import { render, screen } from '@testing-library/react-native';

test('works fine for rendering components from shared package', () => {
  render(<Button />);

  expect(screen).toMatchSnapshot();
});
