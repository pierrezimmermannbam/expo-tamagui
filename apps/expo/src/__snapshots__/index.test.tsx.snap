// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works fine for rendering components from shared package 1`] = `
<View
  accessibilityRole="button"
  cursor="pointer"
  focusable={true}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "$background",
      "borderBottomColor": "transparent",
      "borderBottomLeftRadius": 9,
      "borderBottomRightRadius": 9,
      "borderBottomWidth": 1,
      "borderLeftColor": "transparent",
      "borderLeftWidth": 1,
      "borderRightColor": "transparent",
      "borderRightWidth": 1,
      "borderStyle": "solid",
      "borderTopColor": "transparent",
      "borderTopLeftRadius": 9,
      "borderTopRightRadius": 9,
      "borderTopWidth": 1,
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "height": 44,
      "justifyContent": "center",
      "paddingLeft": 18,
      "paddingRight": 18,
    }
  }
/>
`;
